src/regression_tracker.py

```diff
@@ -107,14 +107,16 @@ def _detect_regression(self, fail_node):
                 fail_node['data']['kernel_revision']['branch'],
             'data.kernel_revision.url':
                 fail_node['data']['kernel_revision']['url'],
-            'data.arch': fail_node['data']['arch'],
-            'data.defconfig': fail_node['data']['defconfig'],
-            'data.config_full': fail_node['data']['config_full'],
-            'data.compiler': fail_node['data']['compiler'],
-            'data.platform': fail_node['data']['platform'],
             'created__lt': fail_node['created'],
-            'state': 'done'
-        })
+            'state': 'done',
+            # Parameters that may be null in some test nodes
+            'data.arch': fail_node['data'].get('arch'),
```

Comment:
API queries with `field=None` are not supported yet.
```
$ curl 'https://staging.kernelci.org:9000/latest/nodes?data.debug=None&limit=1'
{"items":[],"total":0,"limit":1,"offset":0}
```
```
$ curl 'https://staging.kernelci.org:9000/latest/nodes?data.debug=null&limit=1'
{"items":[{"id":"65a12eabe98651d0fe81e3ed","kind":"checkout","name":"checkout","path":["checkout"],"group":null,"parent":null,"state":"done","result":null,"artifacts":null,"data":{"kernel_revision":{"tree":"kernelci","url":"https://github.com/kernelci/linux.git","branch":"staging-mainline","commit":"3145cd9dd8e634f31e1e14cb4f02d23573354585"}},"debug":null,"created":"2024-01-12T12:20:59.469000","updated":"2024-01-12T13:21:55.301000","timeout":"2024-01-12T13:20:59.406000","holdoff":null,"owner":"admin","user_groups":[]}],"total":1093301,"limit":1,"offset":0}
```

Reply:
As per the default behavior of `FastAPI`, all the query params are sent as `str` to API.
API translates them later based on the field type. Hence, the API will receive `'None'` and not `None` from the above query.

> But I'll set them to null explicitly so that the intention is more explicit.

Bdw this would also need a translation method of query param values with `null` as if you send `'null'` directly in the query, it will be again sent as `str`.

Reply:
OK. I tried and API is working with string `'null'`.